tests/scalers/couchdb/couchdb_test.go

```diff
@@ -0,0 +1,394 @@
+	if err != nil {
+			fmt.Println("Failed to create test pod: ", err)
+	}
+}
+
+func testActivation(t *testing.T, kc *kubernetes.Clientset) {
+	t.Log("--- testing activation ---")
+	password := getPassword(kc, testNamespace)
+	if WaitForStatefulsetReplicaReadyCount(t, kc, "test-release-couchdb", testNamespace, 3, 15, 1) {
+		dsn := fmt.Sprintf("http://admin:%s@test-release-svc-couchdb.%s.svc.cluster.local:5984/animals", password, testNamespace)
+		cmd := "-vX PUT " + dsn
+		fmt.Println("Splitted command: ", strings.Split(cmd, " "))
+		deployPod(kc, "activation", cmd)
+		AssertReplicaCountNotChangeDuringTimePeriod(t, kc, deploymentName, testNamespace, minReplicaCount, 5)
+	}
+}
+
+func insertRecord(kc *kubernetes.Clientset, password, record, name, uuid string) {
+	dsn := fmt.Sprintf(`http://admin:%s@test-release-svc-couchdb.%s.svc.cluster.local:5984/animals/%s`, password, testNamespace, uuid)
+	cmd := ` -X PUT ` + dsn + ` -d ` + record
+	fmt.Println(cmd)
+	deployPod(kc, name, cmd)
+}
+
+func deployPodDelete(kc *kubernetes.Clientset, podName, args string) {
+	podSpec := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-pod-couchdb-" + podName,
+			Namespace: testNamespace,
+		},
+		Spec: corev1.PodSpec{
+			RestartPolicy: "OnFailure",
+			Containers: []corev1.Container{
+				{
+					Name:    "test-couchdb-container",
+					Image:   "nginx:alpine",
+					Command: []string{"curl"},
+					Args:    strings.Split(args, " "),
+				},
+			},
+		},
+	}
+	_, err := kc.CoreV1().Pods(testNamespace).Create(context.Background(), podSpec, metav1.CreateOptions{})
+	if err != nil {
+			fmt.Println("Failed to create test pod: ", err)
+	}
+}
+
+func deleteRecord(kc *kubernetes.Clientset, password, name , uuid, rev string) {
+	fmt.Printf("Rev id: %s-test", rev)
+	fmt.Println()
+	dsn := fmt.Sprintf("http://admin:%s@test-release-svc-couchdb.%s.svc.cluster.local:5984/animals/%s?rev=%s", password, testNamespace, uuid, rev)
+	cmd := " -X DELETE " + dsn 
+	fmt.Println("curl" + cmd)
+	fmt.Println("curl" + cmd)
+	deployPodDelete(kc, name, cmd)
+}
+
+type PodLog struct {
+    OK  bool   `json:"ok"`
+	ID  string `json:"id"`
+	Rev string `json:"rev"`
+}
+
+func getPodLogs(kc *kubernetes.Clientset, podName string) (string, error) {
+	podLogOpts := corev1.PodLogOptions{}
+	req := kc.CoreV1().Pods(testNamespace).GetLogs(podName, &podLogOpts)
+	podLogs, err := req.Stream(context.Background())
+	if err != nil {
+			fmt.Println("error in opening stream: ", err)
+	return "", err
+	}
+	defer podLogs.Close()
+
+	buf := new(bytes.Buffer)
+	_, err = io.Copy(buf, podLogs)
+	if err != nil {
+			fmt.Println("error in copy information from podLogs to buf: ", err)
+	return "", err
+	}
+	str := buf.String()
+	fmt.Println("Index: ", strings.Index(str, `"rev":`))
+	chars := str[strings.Index(str, `"rev":`)+7 : strings.Index(str, `}`)-1]
+	fmt.Println(chars)
+	fmt.Println("Type of chars: ", reflect.TypeOf(chars))
+	return chars, nil
+}
```

Comment:
There is a helper for this https://github.com/kedacore/keda/blob/main/tests/helper/helper.go#L507

Reply:
It's a customised pod log function, I am trying to get specific lines with some string ops